### PR TITLE
Add IoU metric to QR code benchmark

### PR DESCRIPTION
**Pull request by AleksandrPanov**

### CODE REVIEW

File: python_benchmarks/qr_codes/qr.py [hunk 1]
```diff
@@ -33,19 +35,28 @@ def __init__(self):
...
-        return r, decoded_info, straight_qrcode
+    def detect(self, image, use_alignment=True):
+            _, decoded_info, corners, _ = self.detector.detectAndDecodeMulti(image)
+            if corners is None or len(corners) == 0:
+                return False, np.array([])
+            self.decoded_info = decoded_info
+            self.detected_corners = corners
+            return True, corners
+
+    def detect_aruco(self, image):
```
Comment (asmorkalov) [inline python_benchmarks/qr_codes/qr.py]: Not used in code.

File: python_benchmarks/qr_codes/qr.py [hunk 2]
```diff
@@ -162,11 +181,11 @@ def main():
                         default="./", action="store",
                         dest="model_path")
     parser.add_argument("-a", "--accuracy", help="input accuracy", default="20", action="store", dest="accuracy",
-                        type=int)
+                        type=float)
     parser.add_argument("-alg", "--algorithm", help="QR detect algorithm", default="opencv", action="store",
```
Comment (asmorkalov) [inline python_benchmarks/qr_codes/qr.py:161]: It makes sense to extend QrCode class with read/write methods to dump settings as it's done for cv::Algorithm implementations and feed config with command line. It allows to select implementations from objdetect and try custom settings.

File: python_benchmarks/qr_codes/qr.py [hunk 3]
```diff
@@ -33,19 +35,28 @@ def __init__(self):
...
+            self.detected_corners = corners
+            return True, corners
+
+    def detect_aruco(self, image):
+        self.detector.detectMultiAruco(image)
+
+    def decode(self, image, use_alignment=True):
+        if use_alignment:
+            if len(self.decoded_info) == 0:
+                return 0, [], None
```
Comment (asmorkalov) [inline python_benchmarks/qr_codes/qr.py]: This part is the same for both `use_aligment` values and should be extracted.

File: python_benchmarks/qr_codes/qr.py [hunk 4]
```diff
@@ -33,19 +35,28 @@ def __init__(self):
...
+            return True, self.decoded_info, self.detected_corners
```
Comment (asmorkalov) [inline python_benchmarks/qr_codes/qr.py]: It does not run decode actually. Is it expected?

File: python_benchmarks/qr_codes/qr.py [hunk 5]
```diff
@@ -188,17 +213,24 @@ def main():
...
         qr_decode = 0
+        category_dist = 0
+        category_pos_dist = 0
         for img_path in imgs_path:
             label_path = img_path[:-3] + "txt"
             gold_corners = get_gold_corners(label_path)
             qr_count += gold_corners.shape[0]
             image = cv.imread(img_path, cv.IMREAD_IGNORE_ORIENTATION)
+            empty_img = np.zeros(image.shape, dtype=np.uint8)
+            empty_img2 = np.zeros(image.shape, dtype=np.uint8)
```
Comment (asmorkalov) [inline python_benchmarks/qr_codes/qr.py]: Are the images detection masks? Please name them accordingly. Do we need them outside of `get_norm_to_rotate_qr`?

**Review by asmorkalov — CHANGES_REQUESTED** [review]

File: python_benchmarks/qr_codes/qr.py (hunk 5 above)
Comment (AleksandrPanov) [reply python_benchmarks/qr_codes/qr.py]: rename `empty_img` to `mask`
remove outside  variable declaration

File: python_benchmarks/qr_codes/qr.py (hunk 4 above)
Comment (AleksandrPanov) [reply python_benchmarks/qr_codes/qr.py]: removed

File: python_benchmarks/qr_codes/qr.py (hunk 3 above)
Comment (AleksandrPanov) [reply python_benchmarks/qr_codes/qr.py]: removed

File: python_benchmarks/qr_codes/qr.py (hunk 1 above)
Comment (AleksandrPanov) [reply python_benchmarks/qr_codes/qr.py]: removed

File: python_benchmarks/qr_codes/qr.py (hunk 2 above)
Comment (AleksandrPanov) [reply python_benchmarks/qr_codes/qr.py:161]: I think it's better to add to the new PR.

File: python_benchmarks/qr_codes/qr.py [hunk 6]
```diff
@@ -146,7 +156,6 @@ def read_output(path):
             decoded_info = image_info.getNode("decoded_info")
             if not decoded_info.empty():
                 for i in range(decoded_info.size()):
-                    # print(decoded_info.at(i).string())
                     pass
```
Comment (asmorkalov) [inline python_benchmarks/qr_codes/qr.py]: Please remove it at all.

Comment (AleksandrPanov) [reply python_benchmarks/qr_codes/qr.py]: removed

**Review by asmorkalov — APPROVED** [review]

**Pull request merged**